src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = (props) => 
+{
+  const onHeartButtonClick = () => {
+    const updatedChat = {
+        id: props.id,
+        sender: props.sender,
+        body: props.body,
+        timeStamp : props.timeStamp,
+        liked: !props.liked
+    };
+      props.onUpdate(updatedChat);
+    };
```

Comment:
While this approach is consistent with what's shown in Learn, it has the downside that a presentation component needs to have enough information to reconstruct an object (including data not required for display) and know how to mutate the new object in the process.

I prefer the approach where only the object id is passed to the event handler. Sending up an id allows the child component to stay simpler, as it only needs the id and any data required for actual display, leaving all the messy stuff to be handled by code that has more knowledge of what the full data might look like.